src/components/Layout/Header/NavBar/KeepKey/ChangePin.tsx

```diff
@@ -1,50 +1,139 @@
 import { useKeepKey } from 'context/WalletProvider/KeepKeyProvider'
 import { useWallet } from 'hooks/useWallet/useWallet'
 import { logger } from 'lib/logger'
 
+import { useMenuRoutes } from '../hooks/useMenuRoutes'
 import { SubMenuBody } from '../SubMenuBody'
 import { SubMenuContainer } from '../SubMenuContainer'
+import { LastDeviceInteractionStatus } from './LastDeviceInteractionStatus'
 
 const moduleLogger = logger.child({
   namespace: ['Layout', 'Header', 'NavBar', 'KeepKey', 'ChangePin'],
 })
 
 export const ChangePin = () => {
+  const { handleBackClick } = useMenuRoutes()
   const translate = useTranslate()
   const { keepKeyWallet } = useKeepKey()
   const {
+    dispatch,
     state: {
-      deviceState: { awaitingDeviceInteraction },
+      keepKeyPinRequestType,
+      deviceState: { awaitingDeviceInteraction, isUpdatingPin, isLoading },
     },
+    setDeviceState,
   } = useWallet()
   const toast = useToast()
+  const pinButtonsBackground = useColorModeValue('gray.200', 'gray.600')
+  const pinButtonsBackgroundHover = useColorModeValue('gray.100', 'gray.500')
+
+  const translationType = (() => {
+    switch (keepKeyPinRequestType) {
+      case PinMatrixRequestType.NEWFIRST:
+        return 'newPin'
+      case PinMatrixRequestType.NEWSECOND:
+        return 'newPinConfirm'
+      default:
+        return 'pin'
+    }
+  })()
+
+  const handleCancel = async () => {
+    const fnLogger = moduleLogger.child({ namespace: ['handleChangePinBackClick'] })
+
+    await keepKeyWallet?.cancel().catch(e => {
+      fnLogger.error(e, 'Error cancelling new PIN...')
+      toast({
+        title: translate('common.error'),
+        description: e?.message?.message ?? translate('common.somethingWentWrong'),
+        status: 'error',
+        isClosable: true,
+      })
+    })
+
+    setDeviceState({
+      isUpdatingPin: false,
+    })
+  }
+
+  const handleHeaderBackClick = async () => {
+    await handleCancel()
+
+    handleBackClick()
+  }
 
   const handleChangePin = async () => {
     const fnLogger = moduleLogger.child({ namespace: ['handleChangePin'] })
     fnLogger.trace('Applying new PIN...')
 
+    setDeviceState({
+      isUpdatingPin: true,
+      awaitingDeviceInteraction: true,
+    })
+
+    dispatch({ type: WalletActions.RESET_LAST_DEVICE_INTERACTION_STATE })
+
     await keepKeyWallet?.changePin().catch(e => {
       fnLogger.error(e, 'Error applying new PIN')
       toast({
         title: translate('common.error'),
-        description: e?.message ?? translate('common.somethingWentWrong'),
+        description: e?.message?.message ?? translate('common.somethingWentWrong'),
         status: 'error',
         isClosable: true,
       })
     })
 
+    setDeviceState({
+      isUpdatingPin: false,
+    })
+
     fnLogger.trace('PIN Changed')
   }
   const setting = 'PIN'
 
+  const shouldDisplayPinView = isUpdatingPin && !awaitingDeviceInteraction
+
   const renderPinState: JSX.Element = (() => {
```

Comment:
I don't know what I was thinking when I named this initially, but because it's an IIFE, it's rendered statically, and so should actually be `renderedPinState`.

```suggestion
  const renderedPinState: JSX.Element = (() => {
```